scripts/lib/CIME/case/case.py

```diff
@@ -577,8 +580,8 @@ def _valid_compset_impl(self, compset_name, comp_classes, comp_hash):
                 model_set[comp_ind] = stub
 
         # Return the completed compset
-        if bgc is not None:
-            model_set.append(bgc)
+        for noncomp in noncomps:
+            model_set.append(noncomp)
```

Comment:
I realize that this follows my previous methodology but I am not sure that BGC or TEST belong in the model set.
I think they should just be appended to the compset string.
Also, I thought there was no use of the TEST appendage, why not just eliminate it from all compsets?

Reply:
There is a usage for DTEST and other compsets which want to minimize inputdata requirements for testing.  I am open to a better long term solution if you have one.

Reply:
I think we should take out these two lines (but see below).